[router] Remove redundant `root` fallback in `useNavigation` (#48638)
expo/expo: packages/expo-router/CHANGELOG.md

### 🛠 Breaking changes

> - Remove `NavigationContainerRefContext` fallback in `useNavigation`. The hook now throws when called outside a navigator, including components rendered via `ExpoRoot`'s `wrapper` prop. ([#48638](https://github.com/expo/expo/pull/48638) by [@Ubax](https://github.com/Ubax))

Author: ```suggestion
- Remove root container fallback from `useNavigation`. The hook now throws when called outside a screen or navigator. ([#48638](https://github.com/expo/expo/pull/48638) by [@Ubax](https://github.com/Ubax))
```
